### ISSUE TITLE
Config Field Sync: Frontend/Backend Mismatches

### ISSUE THREAD

**Issue by duskfallcrew** [Ktiseos-Nyx/Ktiseos-Nyx-Trainer#97]
## 📊 Config Field Synchronization Tracking

During verification, we found some fields that exist in one layer but not the other. **This is not causing failures** - the system works correctly, but we should track this for future completeness.

---

## ⚠️ Fields in Frontend BUT NOT Backend (16 fields)

These fields are defined in `frontend/lib/api.ts` but **missing from `services/models/training.py`**.

**Impact:** Values sent from frontend are silently dropped by Pydantic validation. No errors, but features don't work.

### Missing Fields:
```typescript
bucket_reso_steps?: number;
cache_text_encoder_outputs_to_disk?: boolean;
full_bf16?: boolean;
max_data_loader_n_workers?: number;
min_timestep?: number;
max_timestep?: number;
multires_noise_iterations?: number;
ip_noise_gamma_random_strength?: boolean;
scale_v_pred_loss_like_noise_pred?: boolean;
v_pred_like_loss?: number;
debiased_estimation_loss?: boolean;
save_state_on_train_end?: boolean;
resume_from_state?: string;
log_tracker_name?: string;
log_tracker_config?: string;
wandb_run_name?: string;
```

**Categorization:**
- **Performance/Memory:** `cache_text_encoder_outputs_to_disk`, `full_bf16`, `max_data_loader_n_workers`
- **Advanced Training:** `min_timestep`, `max_timestep`, `multires_noise_iterations`, `ip_noise_gamma_random_strength`
- **Loss Functions:** `scale_v_pred_loss_like_noise_pred`, `v_pred_like_loss`, `debiased_estimation_loss`
- **State Management:** `save_state_on_train_end`, `resume_from_state`
- **Logging:** `log_tracker_name`, `log_tracker_config`, `wandb_run_name`
- **Bucketing:** `bucket_reso_steps`

---

## ⚠️ Fields in Backend BUT NOT Frontend (11 fields)

These fields exist in `services/models/training.py` but **not exposed in frontend UI**.

**Impact:** Users can't configure these from the UI. Backend uses default values.

### Missing Fields:
```python
training_mode: TrainingMode  # Default: LORA
network_train_unet_only: bool  # Default: True
prior_loss_weight: float  # Default: 1.0

# Block-wise learning rates (advanced)
down_lr_weight: Optional[str]  # Default: None
mid_lr_weight: Optional[str]  # Default: None
up_lr_weight: Optional[str]  # Default: None
block_lr_zero_threshold: Optional[str]  # Default: None
block_dims: Optional[str]  # Default: None
block_alphas: Optional[str]  # Default: None
conv_block_dims: Optional[str]  # Default: None
conv_block_alphas: Optional[str]  # Default: None
```

**Categorization:**
- **Mode Selection:** `training_mode` (LoRA vs Full checkpoint)
- **Training Scope:** `network_train_unet_only`, `prior_loss_weight`
- **Advanced LyCORIS:** All the block-wise LR fields (power user features)

---

## ✅ What's Working Correctly

All **essential training parameters** are properly synchronized:
- ✅ Project setup (model, VAE, paths)
- ✅ Dataset config (resolution, batch size, epochs)
- ✅ Learning rates & schedulers
- ✅ LoRA structure (all 11 types now supported!)
- ✅ Optimizer settings
- ✅ Memory & performance
- ✅ Saving & checkpoints
- ✅ Model-specific (Flux, SD3, Lumina)

---

### Phase 1: Add Missing Backend Fields
Add the 16 frontend fields to `services/models/training.py`:
- Update Pydantic model with all missing fields
- Map to TOML generation in `kohya_toml.py`
- Test that sd-scripts accepts these parameters

### Phase 2: Add Missing Frontend Fields
Expose the 11 backend fields in UI:
- Add to `frontend/lib/api.ts` TypeScript interface
- Create UI components for advanced settings
- Document block-wise LR usage (advanced users only)

### Phase 3: Validation
- Add tests to ensure frontend/backend stay in sync
- Consider generating TypeScript types from Pydantic models

---

## 📝 Current Status

**Decision:** Keep as-is for now (Option A)
- No user-facing failures
- All essential features work correctly
- Can be improved incrementally

**Priority:** Low - Enhancement/Technical Debt
**Milestone:** Post-Alpha cleanup

---

## 🔗 Related

- Fixed in #83 - onSave functionality
- Related to training config validation system
- May benefit from schema generation tools

---

**Verified:** 2026-01-02
**Files Checked:**
- `frontend/lib/api.ts` (TrainingConfig interface)
- `services/models/training.py` (TrainingConfig Pydantic model)
- `services/trainers/kohya_toml.py` (TOML generation)
